### ISSUE TITLE
Ooops! kvrocks version 2.6.0 (commit cac0ade4) got signal: Segmentation fault (11)

### ISSUE THREAD

**Issue by iushas** [apache/kvrocks#1908]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Version

2.6.0

### Minimal reproduce step

-

### What did you expect to see?

-

### What did you see instead?
```
root@host:/home/redis/var/log# cat kvrocks.ERROR
Log file created at: 2023/11/24 10:45:56
Running on machine: redis-kvsenior01-210027.i.nease.net
Running duration (h:mm:ss): 0:00:00
Log line format: [IWEF]yyyymmdd hh:mm:ss.uuuuuu threadid file:line] msg
E20231124 10:45:56.856230  5147 main.cc:85] ======= Ooops! kvrocks version 2.6.0 (commit cac0ade4) got signal: Segmentation fault (11) =======
E20231124 10:45:56.856793  5147 main.cc:102] /lib/x86_64-linux-gnu/libpthread.so.0(+0x14140) [0x7f45cb08d140]
E20231124 10:45:56.857265  5147 main.cc:100] /home/redis/server/2.6.0/bin/kvrocks(+0x217769) [0x560584630769]  redis::Connection::ExecuteCommands()
E20231124 10:45:56.857934  5147 main.cc:100] /home/redis/server/2.6.0/bin/kvrocks(+0x1605ac) [0x5605845795ac]  EvbufCallbackBase<>::readCB()
E20231124 10:45:56.858131  5147 main.cc:100] /home/redis/server/2.6.0/bin/kvrocks(+0x7b0dee) [0x560584bc9dee]  bufferevent_run_deferred_callbacks_unlocked
E20231124 10:45:56.858312  5147 main.cc:100] /home/redis/server/2.6.0/bin/kvrocks(+0x7b738d) [0x560584bd038d]  event_process_active_single_queue
E20231124 10:45:56.858755  5147 main.cc:100] /home/redis/server/2.6.0/bin/kvrocks(+0x7b7d4f) [0x560584bd0d4f]  event_base_loop
E20231124 10:45:56.859345  5147 main.cc:100] /home/redis/server/2.6.0/bin/kvrocks(+0x23af1d) [0x560584653f1d]  _ZNSt6thread11_State_implINS_8_InvokerISt5tupleIJZN4util12CreateThreadIZN12WorkerThread5StartEvEUlvE_EE8StatusOrIS_EPKcT_EUlvE_EEEEE6_M_runEv.lto_priv.0
E20231124 10:45:56.859525  5147 main.cc:100] /home/redis/server/2.6.0/bin/kvrocks(+0x9aa7b0) [0x560584dc37b0]  execute_native_thread_routine
E20231124 10:45:56.859648  5147 main.cc:100] /lib/x86_64-linux-gnu/libpthread.so.0(+0x8ea7) [0x7f45cb081ea7]   start_thread
E20231124 10:45:56.859750  5147 main.cc:100] /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f45cae6ddef]      clone

```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by iushas** [comment]
-  addr2line


```
root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks +0x217769
/home/redis/kvrocks/src/server/server.cc:1505
```
![image](https://github.com/apache/kvrocks/assets/7236171/1d8d5466-192d-483e-85f4-f23583cd2d6f)

```
root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks 0x23af1d
/home/redis/kvrocks/src/server/worker.cc:292
```
![image](https://github.com/apache/kvrocks/assets/7236171/05cc8b92-289f-4f9a-a01a-08024c2184ff)


```
root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks +0x9aa7b0
thread.o:?

root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks 0x9aa7b0
thread.o:?

root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks 0x23af1d
/home/redis/kvrocks/src/server/worker.cc:292


root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks 0x7b7d4f
/home/redis/kvrocks/build/_deps/libevent-src/event.c:1790


root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks +0x217769
/home/redis/kvrocks/src/server/server.cc:1505


root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks +0x1605ac
/usr/include/c++/10/bits/atomic_base.h:426

root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks +0x7b0dee
/home/redis/kvrocks/build/_deps/libevent-src/bufferevent.c:208 (discriminator 3)

root@host:/home/redis/var/log# addr2line -e  /home/redis/server/2.6.0/bin/kvrocks +0x7b7d4f
/home/redis/kvrocks/build/_deps/libevent-src/event.c:1790

root@host:/home/redis/var/log# addr2line -e /lib/x86_64-linux-gnu/libpthread.so.0 +0x14140
??:?
```

- dmesg
```
[Fri Nov 24 10:42:27 2023] worker[3173205]: segfault at 8 ip 0000560584630769 sp 00007f45bd7ec420 error 4 in kvrocks[56058449f000+94f000]
[Fri Nov 24 10:42:27 2023] Code: 8b 54 24 68 41 8b 44 24 68 c1 e8 06 81 e2 00 01 00 00 89 95 78 fe ff ff 0f 94 c2 84 c2 0f 85 db 0e 00 00 48 8b 85 f0 fe ff ff <48> 83 78 08 00 0f 85 bc 01 00 00 4c 8d 75 b0 4c 8d 7d c0 be 08 00
```

**Comment by git-hulk** [comment]
@iushas Thanks for your report. This will be fixed in #1909.
